### DISCUSSION TITLE
`use-node-version` in `.npmrc` fails deployment on vercel

### DISCUSSION THREAD

**Discussion by stefanprobst** [vercel/community#2436] [Help] [answered]
### Problem Description

when a `.npmrc` file includes `use-node-version=18.16.0` (see https://pnpm.io/npmrc#use-node-version), deployment on vercel fails with

![Screenshot_20230511_192907](https://github.com/vercel/vercel/assets/20753323/fc6bcb27-be2a-4f63-a2bd-8a7a50a35e1d)

you can check by trying to deploy https://github.com/stefanprobst/issue-vercel-npmrc

**Comment by amyegan** [comment, marked as answer]
Vercel's [Node.js Runtime](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js) currently supports major version `18.x`, `16.x`, and `12.x`. Minor and patch updates are automatically rolled out as needed to address security issues and bugs, so only major versions can be specified.

You can control which of these major versions is used by selecting one under [Node.js Version](https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#node.js-version) on the project settings page or by defining the `node` property inside [`engines`](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#engines) of the project's `package.json` file.

**Reply by stefanprobst** [reply]
thanks for your response!
i am aware that on vercel i can specify the used major node version (and have it set to 18.x via `engines`) -- however i am using the `use-node-version` setting in `.npmrc` like a nvm replacement in local development, and i think having this setting in `.npmrc` should not fail deployments to vercel with a cryptic error message about glibc.

**Reply by ericmatthys** [reply]
I've also run into this error. `use-node-version` is enticing as a way to remove the need for an extra tool like volta or nvm in local development, but it's affecting Vercel builds. It seems like the install of Node.js that pnpm is swapping in when running pnpm commands during the build is not compatible in the build environment for some reason. To work around it, it seems like we'd need to run a script during builds to remove this setting from `.npmrc` before trying to install dependencies.

**Reply by swarnava** [reply]
Hi Eric,

I have passed this feedback to relevant team so that we can ignore `use-node-version` in future. For now, I will recommend not to set the field or setup a `.vercelignore` file to ignore `.npmrc` file during deployment.

**Reply by ericmatthys** [reply]
Thanks. There are other settings in `.npmrc` that are important to respect when installing dependencies, so for now we're just not using this feature of pnpm.

**Reply by stefanprobst** [reply]
looks like this has been addressed in https://github.com/vercel/vercel/pull/10049 by suggesting to use "engines" instead of "use-node-version", not by ignoring the "use-node-version" setting on vercel
